File: 283. Move Zeroes/solution.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        zeroes = 0

        for index, num in enumerate(nums):

            if num == 0:
                zeroes += 1
            else:
                nums[index-zeroes] = num

        nums[len(nums)-zeroes:] = [0] * zeroes

File: 283. Move Zeroes/Solution.cs
public class Solution {
    public void MoveZeroes(int[] nums) {
        
        int zeroes = 0;

        for (int i = 0; i < nums.Length; ++i) {

            if (nums[i] == 0) {
                zeroes++;
            }
            else {
                nums[i-zeroes] = nums[i];
            }
        }

        for (int i = nums.Length-zeroes; i < nums.Length; ++i) {
            nums[i] = 0;
        }
    }
}
